### DISCUSSION TITLE
Any way to pre-decode spectogram?

### DISCUSSION THREAD

**Discussion by kerojohan** [katspaugh/wavesurfer.js#3792] [Q&A]
Hi!

I'm working with very large files, since today I was pre calculating peaks to see the waveform, it works like a charm, now I want to add a spectrogram but does not seem possible rendering it with the browser, is there any way to pre generate spectograms and load this data to wavesurfer?

Thanks!

**Comment by katspaugh** [comment]
You should check the source code of the spectrogram plugin, I think there was a method to load arbitrary FFT data.

**Reply by kerojohan** [reply]
Thanks katspaugh,
seems that is not so easy, I've seen questions about it since 2018 but without any clear solution, wafesurfer is not able to create a Bitmap moving such quantity of information, would be great to load pre-calculated images done with python with librosa lib.

Anyone had the same need or problem and could be able to make a workaround?

Thanks!

**Comment by melijimenez11** [comment]
@katspaugh @kerojohan 

I'm having the same problem. Working with very large files, and it's impossible to generate the spectrogram client-side. Any help would be appreciated!

**Reply by katspaugh** [reply]
Documented it here: https://github.com/katspaugh/wavesurfer.js/commit/0ffead9ed1cefacc732de46256bde69694aa30cf
